microprofile/microprofile-health-check-2.0/src/test/java/org/wildfly/swarm/ts/microprofile/health/check/v20/MicroProfileHealthCheck20Test.java

```diff
@@ -0,0 +1,80 @@
+package org.wildfly.swarm.ts.microprofile.health.check.v20;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import org.apache.http.client.fluent.Request;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.wildfly.swarm.arquillian.DefaultDeployment;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@RunWith(Arquillian.class)
+@DefaultDeployment
+public class MicroProfileHealthCheck20Test {
+    private static final String URL_HEALTH = "http://localhost:8080/health";
+
+    private static final String URL_HEALTH_LIVE = URL_HEALTH + "/live";
+
+    private static final String URL_HEALTH_READY = URL_HEALTH + "/ready";
+
+    @Test
+    @RunAsClient
+    public void test() throws IOException {
+        String response = Request.Get(URL_HEALTH).execute().returnContent().asString();
+        // expected check count is 5 because BothHealthCheck contains both annotations: @Liveness and @Readiness
+        checkResponse(response, 5, Arrays.asList(BothHealthCheck.class,
+                                                 DeprecatedHealthCheck.class,
+                                                 LivenessHealthCheck.class,
+                                                 ReadinessHealthCheck.class
+        ));
+
+        response = Request.Get(URL_HEALTH_LIVE).execute().returnContent().asString();
+        checkResponse(response, 2, Arrays.asList(BothHealthCheck.class,
+                                                 LivenessHealthCheck.class
+
+        ));
+
+        response = Request.Get(URL_HEALTH_READY).execute().returnContent().asString();
+        checkResponse(response, 2, Arrays.asList(BothHealthCheck.class,
+                                                 ReadinessHealthCheck.class
+        ));
+    }
+
+    private void checkResponse(String response, int checkCount, List<Class> names) {
```

Comment:
Instead of `List<Class>` for the set of expected names, I'd rather have `Set<String>`.